components/SponsoredFeedsTableWithData.tsx

```diff
@@ -22,56 +20,82 @@ interface SponsoredFeedsTableProps {
 /**
  * Helper functions
  */
+// Convert time_difference (seconds) to human readable format
+const formatTimeUnit = (seconds: number): { value: number; unit: string } => {
+  if (seconds >= 3600) {
+    return { value: seconds / 3600, unit: "hour" };
+  } else if (seconds >= 60) {
+    return { value: seconds / 60, unit: "minute" };
+  } else {
+    return { value: seconds, unit: "second" };
+  }
+};
 
 // Format update parameters as a string for grouping
-const formatUpdateParams = (params: UpdateParameters): string => {
-  return `${params.heartbeatLength} ${params.heartbeatUnit} heartbeat / ${params.priceDeviation}% price deviation`;
+const formatUpdateParams = (feed: SponsoredFeed): string => {
+  const timeFormat = formatTimeUnit(feed.time_difference);
+  const timeStr = `${timeFormat.value} ${timeFormat.unit}${
+    timeFormat.value !== 1 ? "s" : ""
+  }`;
+  return `${timeStr} heartbeat / ${feed.price_deviation}% price deviation`;
 };
 
 // Render update parameters with proper styling
-const renderUpdateParams = (params: UpdateParameters, isDefault: boolean) => (
-  <div className="flex items-start gap-1.5">
-    <div
-      className={`w-1.5 h-1.5 rounded-full mt-1 flex-shrink-0 ${
-        isDefault ? "bg-green-500" : "bg-orange-500"
-      }`}
-    ></div>
-    <span
-      className={`text-xs leading-relaxed font-medium ${
-        isDefault
-          ? "text-gray-700 dark:text-gray-300"
-          : "text-orange-600 dark:text-orange-400"
-      }`}
-    >
-      <strong>{params.heartbeatLength}</strong> {params.heartbeatUnit} heartbeat
-      <br />
-      <strong>{params.priceDeviation}%</strong> price deviation
-    </span>
-  </div>
-);
+const renderUpdateParams = (feed: SponsoredFeed, isDefault: boolean) => {
```

Comment:
This should be a component since it returns jsx, and the only change would be renaming it from `renderUpdateParams` to `UpdateParams`, and changing the arguments to an object.

By using a component, React can optimize rerenders and the component tree.  By using a function, react is not aware of this subtree as a component and applies no render optimizations.